packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -321,6 +321,10 @@ export class TerminalFrontendContribution implements FrontendApplicationContribu
         return this.widgetManager.getWidgets(TERMINAL_WIDGET_FACTORY_ID) as TerminalWidget[];
     }
 
+    get allVisible(): TerminalWidget[] {
+        return this.all.filter(widget => !widget.hidden);
```

Comment:
Since widgets can be programmatically hidden in other ways, it may make sense to use the `isHidden` property here, as well. In fact, it might have made sense from the start, to offer only those widgets in the quick pick :-).

Reply:
@planger I believe this comment is not addressed?

There are two types of **hidden** at play here which can be confusing for downstream adopters. A widget can be hidden manually through the UI, or may not have been shown yet (`hideFromUser`). For adopters they may be using `allVisible` and wondering why they are given terminals which are hidden by phosphor.